### PR TITLE
MPAS native output support + example

### PR DESCRIPTION
**Pull request by zmoon**
* unstructured grid support in `contours` and `identify`, assumed if only one spatial dimension
  - Using Delaunay triangulation on lat/lon grid points (or user-provided `Triangulation`) + Matplotlib `tricontour`
* example with subset of native MPAS output, same time and location as the existing (regridded) MPAS example dataset
* gdown support for downloading examples added as mentioned in #9 
* [x] some comparison to [mpas-tools grid-to-triangles Triangulation](https://mpas-dev.github.io/MPAS-Tools/stable/visualization.html#mpas-mesh-to-triangles)
* some doc updates

> [!NOTE]
> Should add `dask-core` and `gdown` to the conda-forge recipe for this release.

### CODE REVIEW

File: docs/examples/sample-mpas-ug-data.ipynb [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bb5b7f8-ebc6-40e8-b6e2-2614d66dd955",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds_rg = tams.load_example_mpas().sel(lat=slice(None, 20))  # same lat upper bound as in the ug data\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(10, 4), subplot_kw=dict(projection=proj), constrained_layout=True)\n",
```
Comment (zmoon) [inline docs/examples/sample-mpas-ug-data.ipynb:194]: Combine with the other figure (a, b)

**Comment by knubez** [comment]
Directory for sample MPAS native data:
/glade/derecho/scratch/knocasio/FROM_CHEYENNE/15km_MPAS_IFS_8sept_12utc_Thompson_MYNN/diag.2006-09-*.nc"

Directory of mesh:
/glade/derecho/scratch/knocasio/FROM_CHEYENNE/15km_MPAS_model_IFS/mpas_15mesh.nc

File: tams/data.py [hunk 2]
```diff
@@ -66,32 +66,91 @@ def tb_from_ir(r, ch: int):
...
     This dataset contains 6 time steps of 2-hourly data (every 2 hours):
     2006-09-01 00--10
+
+    See Also
+    --------
+    tams.data.download_examples
     """
 
-    ds = xr.open_dataset(HERE / "Satellite_data.nc").rename_dims(
+    ds = xr.open_dataset(HERE / "Satellite_data.nc", lock=False).rename_dims(
```
Comment (zmoon) [inline tams/data.py:153]:
```suggestion
    ds = xr.open_dataset(HERE / "Satellite_data.nc").rename_dims(
```
Just need dask-core, or new-enough xarray that bundles the SerializableLock (v2024.01.1).

File: docs/examples/tracking-options.ipynb [hunk 3]
```diff
@@ -1323,7 +1326,13 @@
     "tags": []
    },
    "source": [
-    "### Identify CEs using different CTT thresholds"
+    "(ctt_thresh_data_in_contours)=\n",
+    "### Identify CEs using different CTT thresholds\n",
+    "\n",
+    "For CEs (not necessarily for MCSs), the CTT threshold only affects identification.\n",
+    "\n",
+    "Here we use {func}`tams.data_in_contours` to add a few cloud-top temperature\n",
```
Comment (knubez) [inline docs/examples/tracking-options.ipynb:1334]: Thanks! <3

**Review by knubez — APPROVED** [review]
Really appreciate all of the descriptive comments and how accessible it is now.

**Pull request merged**